ui/app/registration/directives/patientAction.js

```diff
@@ -138,24 +138,41 @@ angular.module('bahmni.registration')
+                        var checkExists = response.data.results.length;
+                        if (checkExists === 0) {
+                            return false;
+                        } else {
+                            return true;
+                        }
+                    });
+                };
                 var createVisit = function (patientProfileData, forwardUrl) {
                     if (isEmptyVisitLocation()) {
                         $state.go('patient.edit', {patientUuid: $scope.patient.uuid}).then(function () {
                             messagingService.showMessage("error", "NO_LOCATION_TAGGED_TO_VISIT_LOCATION");
                         });
                         return;
                     }
-                    spinner.forPromise($scope.visitControl.createVisitOnly(patientProfileData.patient.uuid, $rootScope.visitLocation).then(function (response) {
-                        auditLogService.log(patientProfileData.patient.uuid, "OPEN_VISIT", {visitUuid: response.data.uuid, visitType: response.data.visitType.display}, 'MODULE_LABEL_REGISTRATION_KEY');
-                        if (forwardUrl) {
-                            var updatedForwardUrl = appService.getAppDescriptor().formatUrl(forwardUrl, {'patientUuid': patientProfileData.patient.uuid});
-                            $window.location.href = updatedForwardUrl;
-                        } else {
-                            goToVisitPage(patientProfileData);
+                    visitExistsCheck(patientProfileData).then(function (exists) {
+                        if (!exists) {
+                            spinner.forPromise($scope.visitControl.createVisitOnly(patientProfileData.patient.uuid, $rootScope.visitLocation).then(function (response) {
+                                auditLogService.log(patientProfileData.patient.uuid, "OPEN_VISIT", { visitUuid: response.data.uuid, visitType: response.data.visitType.display }, 'MODULE_LABEL_REGISTRATION_KEY');
+                                if (forwardUrl) {
+                                    var updatedForwardUrl = appService.getAppDescriptor().formatUrl(forwardUrl, { 'patientUuid': patientProfileData.patient.uuid });
+                                    $window.location.href = updatedForwardUrl;
+                                } else {
+                                    goToVisitPage(patientProfileData);
+                                }
+                            }, function () {
+                                $state.go('patient.edit', { patientUuid: $scope.patient.uuid });
+                            }));
                         }
-                    }, function () {
-                        $state.go('patient.edit', {patientUuid: $scope.patient.uuid});
-                    }));
+                        else {
+                            messagingService.showMessage("error", "VISIT_OF_THIS_PATIENT_AT_SAME_LOCATION_EXISTS");
```

Comment:
You move this logic to top as a guard clause. 

if(exists) return "error"

And then everything else doesn't need brackets.